pkg/network/usm/shared_libraries.go

```diff
@@ -249,27 +250,33 @@ func (w *soWatcher) Start() {
 			return nil
 		}
 
-		// report silently parsing /proc error as this could happen
-		// just exit processes
-		proc, err := process.NewProcess(int32(pid))
+		mapsPath := fmt.Sprintf("%s/%d/maps", w.procRoot, pid)
+		maps, err := os.Open(mapsPath)
 		if err != nil {
 			log.Debugf("process %d parsing failed %s", pid, err)
 			return nil
 		}
-		mmaps, err := proc.MemoryMaps(true)
-		if err != nil {
-			if log.ShouldLog(seelog.TraceLvl) {
-				log.Tracef("process %d maps parsing failed %s", pid, err)
-			}
-			return nil
-		}
-
-		root := fmt.Sprintf("%s/%d/root", w.procRoot, pid)
-		for _, m := range *mmaps {
-			for _, r := range w.rules {
-				if r.re.MatchString(m.Path) {
-					w.registry.register(root, m.Path, uint32(pid), r)
-					break
+		defer maps.Close()
+
+		scanner := bufio.NewScanner(bufio.NewReader(maps))
+
+		cache := make(map[string]struct{})
+		for scanner.Scan() {
+			line := scanner.Text()
+			cols := strings.Fields(line)
+			// ensuring we have exactly 6 elements in the line and 4th element (length) is not zero (indicates it is a
+			// path, and not an anonymous path).
+			if len(cols) == 6 && cols[4] != "0" {
+				if _, exists := cache[cols[5]]; exists {
+					continue
+				}
+				cache[cols[5]] = struct{}{}
+				for _, r := range w.rules {
+					if r.re.MatchString(cols[5]) {
+						root := fmt.Sprintf("%s/%d/root", w.procRoot, pid)
+						w.registry.register(root, cols[5], uint32(pid), r)
+						break
+					}
```

Comment:
Let's please move this to a helper function, with proper documentation and unit tests similar to what we used to have originally: https://github.com/DataDog/datadog-agent/blob/15b841077356a9c4038c29fbd2ac8c7972e8cd96/pkg/network/so/maps.go

Reply:
great suggestion, added